javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/typeinference/ControlFlowLogic.java

```diff
@@ -100,11 +100,11 @@ public boolean continueADoStatement(ContinueStmt continueStmt, DoStmt doStmt) {
     }
 
     private boolean contains(Statement container, Statement contained) {
-        throw new UnsupportedOperationException();
+        throw new UnsupportedOperationException(String.format("%s is not present in the %s", contained, container));
```

Comment:
Be careful here -- the default `toString()` method pretty prints the contents of the node (e.g. if you pass in a method declaration or switch statement or whatever, then there will be a LOT of output). 

Perhaps make use of `container.getClass().getSimpleName()`?